### PR TITLE
Fix/implement config oneclick mall for testing

### PR DESCRIPTION
**Pull request by eugenioLopezRamos**
Add the `configureForTesting` method to Oneclick mall

### CODE REVIEW

File: lib/onepay/Transaction.php [hunk 1]
```diff
@@ -65,16 +66,19 @@ public static function create($shoppingCart, $channel = null, $externalUniqueNum
...
+        $httpCode = $httpResponse->getStatusCode();
+        $responseJson = json_decode($httpResponse->getBody(), true);
+
+        if ($httpCode != 200 && $httpCode != 204) {
             throw new TransactionCreateException('Could not obtain a response from the service', -1);
         }
-        if($httpResponse['responseCode'] != "OK") {
-            throw new TransactionCreateException($httpResponse['responseCode'] . " : " . $httpResponse['description'], -1);
+        if($responseJson['responseCode'] != "OK") {
+            throw new TransactionCreateException($responseJson['responseCode'] . " : " . $responseJson['description'], -1);
```
Comment (leosoto) [inline lib/onepay/Transaction.php]: Is the code useful for the SDK user? If it is, are we forcing him/her to parse a string in order to extract the code?

File: lib/webpay_rest/Options.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    const DEFAULT_INTEGRATION_TYPE = "TEST";
+    const DEFAULT_INTEGRATION_TYPE_URL = "https://webpay3gint.transbank.cl/";
+    const DEFAULT_WEBPAY_PLUS_MALL_COMMERCE_CODE = '597055555535';
+    const DEFAULT_WEBPAY_PLUS_MALL_CHILD_COMMERCE_CODE_1 = '597055555536';
+    const DEFAULT_WEBPAY_PLUS_MALL_CHILD_COMMERCE_CODE_2 = '597055555537';
+    const DEFAULT_DEFERRED_COMMERCE_CODE = '597055555540';
+
+    const DEFAULT_ONECLICK_MALL_COMMERCE_CODE = '597055555541';
+    const DEFAULT_ONECLICK_MALL_CHILD_COMMERCE_CODE_1 = '597055555542';
+    const DEFAULT_ONECLICK_MALL_CHILD_COMMERCE_CODE_2 = '597055555543';
```
Comment (leosoto) [inline lib/webpay_rest/Options.php]: Maybe the "child" commerce codes should be an array? (I don't see where those consts are used, but still)

File: lib/webpay_rest/oneclick/MallInscription.php [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        if ($httpCode != 200 && $httpCode != 204) {
+            $reason = $httpResponse->getReasonPhrase();
+            $message = "Could not obtain a response from the service: $reason (HTTP code $httpCode)";
+            $body = json_decode($httpResponse->getBody(), true);
+
+            if (isset($body["error_message"])) {
+                $tbkErrorMessage = $body["error_message"];
+                $message = "$message. Details: $tbkErrorMessage";
+            }
+            throw new InscriptionStartException($message, -1);
```
Comment (leosoto) [inline lib/webpay_rest/oneclick/MallInscription.php]: What about the error code? If the SDK user wants to handle the errors described in the docs, how can he do it? Just parsing string?

File: lib/webpay_rest/Options.php (hunk 2 above)
Comment (eugenioLopezRamos) [reply lib/webpay_rest/Options.php]: The array comment is good, I'll implement it. The codes are used for the example projects

File: lib/onepay/Transaction.php (hunk 1 above)
Comment (eugenioLopezRamos) [reply lib/onepay/Transaction.php]: On Onepay all errors have status code 200

File: lib/webpay_rest/oneclick/MallInscription.php (hunk 3 above)
Comment (eugenioLopezRamos) [reply lib/webpay_rest/oneclick/MallInscription.php]: This is good, I'll add that

File: lib/webpay_rest/Options.php (hunk 2 above)
Comment (eugenioLopezRamos) [reply lib/webpay_rest/Options.php]: Implemented: https://github.com/TransbankDevelopers/transbank-sdk-php/pull/88

File: lib/webpay_rest/oneclick/MallInscription.php (hunk 3 above)
Comment (eugenioLopezRamos) [reply lib/webpay_rest/oneclick/MallInscription.php]: Implemented: https://github.com/TransbankDevelopers/transbank-sdk-php/pull/86

**Pull request merged**